src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/GetLatestVersionUtility.cs

```diff
@@ -0,0 +1,119 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using NuGet.Common;
+using NuGet.Configuration;
+using NuGet.Protocol;
+using NuGet.Protocol.Core.Types;
+using NuGet.Versioning;
+
+namespace NuGet.CommandLine.XPlat.Utility
+{
+    public static class GetLatestVersionUtility
```

Comment:
Consider naming this the AddPackageCommandUtility.

EvaluateSources is specific to the command itself, rather than getting the latest version specifically. 

I think that's a better class/filename :)